src/py/flwr/client/grpc_adapter_client/connection.py

```diff
@@ -44,7 +44,7 @@ def grpc_adapter(  # pylint: disable=R0913
     Tuple[
         Callable[[], Optional[Message]],
         Callable[[Message], None],
-        Optional[Callable[[], None]],
+        Optional[Callable[[], int]],
```

Comment:
This type must be the same for all connection types:
```suggestion
        Optional[Callable[[], Optional[int]]],
```